src/components/Calander/CalanderComponent.js

```diff
@@ -0,0 +1,300 @@
+import { React, useState, useEffect } from 'react';
+import styled from 'styled-components';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { DateRangePicker } from 'react-date-range';
+import { addDays, subDays } from 'date-fns';
+import { ko } from 'date-fns/esm/locale';
+
+const CalanderComponent = props => {
+  // onclick
+  function onClickAndPostDataToBackend() {
+    savingResData();
+    sendData();
+  }
+
+  //백엔드로 보낼 데이터 정리
+  const savingResData = () => {
+    resStartDate = startDate.toISOString().split('T')[0];
+    resEndDate = endDate.toISOString().split('T')[0];
+    betweenResDate = [];
+    props.getDateRange(resStartDate, resEndDate, betweenResDate);
+    console.log(findUniqElem(betweenResDate, props.excludeDateArray));
+    if (
+      betweenResDate.length >
+      findUniqElem(betweenResDate, props.excludeDateArray).length
+    ) {
+      alert(
+        `${resStartDate} ~ ${resEndDate} 사이에 이미 예약된 날짜가 있어 예약이 불가능합니다.`
+      );
+      setReservedNight(0);
+    } else {
+      alert(`${resStartDate} ~ ${resEndDate} 예약성공`);
+      setReservedNight(betweenResDate.length);
+      console.log(resStartDate, resEndDate, reservedNight, numberOfGuests);
+    }
+    console.log('1');
+  };
+
+  const sendData = () => {
+    fetch(`http://10.58.52.210:3000/reservation/${props.productId}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        userId: 1,
+        checkIn: `${resStartDate}`,
+        checkOut: `${resEndDate}`,
+        guestCount: `${numberOfGuests}`,
+      }),
+    });
+    console.log('2');
+  };
+
+  // Post
+
+  const [startDate, setStartDate] = useState(new Date());
+  const [endDate, setEndDate] = useState(addDays(new Date(), 4));
+  const [numberOfGuests, setNumberOfGuests] = useState(1);
+  const [buttonBoolean, setButtonBoolean] = useState(false);
+  const [reservedNight, setReservedNight] = useState(0);
+  let resStartDate;
+  let resEndDate;
+  let betweenResDate = [];
```

Comment:
이부분은 startDate, endDate로 computed 될수 있는 값이니 이렇게 바꾸는건 어때요
```suggestion
  const [resStartDate] = startDate.toISOString().split('T');
  const [resEndDate] = endDate.toISOString().split('T');
```

Reply:
수정했습니다!